mtags/src/main/scala-3/scala/meta/internal/pc/SymbolPrinter.scala

```diff
@@ -117,31 +144,100 @@ class SymbolPrinter(using ctx: Context) extends RefinedPrinter(ctx) {
+          // Don't show implicit evidence params
+          // e.g.
+          // from [A: Ordering](a: A, b: A)(implicit evidence$1: Ordering[A])
+          // to   [A: Ordering](a: A, b: A): A
+          if (implicitEvidenceParams.contains(param)) Nil
+          else
+            paramLabel(
+              param,
+              implicitEvidencesByTypeParam,
+              shortenedNames
+            ) :: Nil
+        }
+        // Remove empty params
+        if (labels.isEmpty) Nil
+        else labels.iterator :: Nil
       }
-
-      // Remove empty params
-      if (labels.isEmpty) Nil
-      else labels.iterator :: Nil
     }.iterator
+    val paramLabelss = label(methodParams)
+    val extLabelss = label(extParams)
+
+    val returnType = typeString(shortType(gtpe.finalResultType, shortenedNames))
+    val extensionSignature = paramssString(extLabelss, extParams)
+
+    val extensionSignatureString =
+      if (extParams.nonEmpty) extensionSignature.mkString("extension ", "", " ")
+      else ""
 
-    val returnType =
-      typeString(shortType(gtpe.finalResultType, shortenedNames))
-    methodSignature(paramLabelss, paramss, returnType)
+    val paramssSignature = paramssString(paramLabelss, methodParams)
+      .mkString("", "", s": ${returnType}")
+
+    if (onlyMethodParams) paramssSignature
+    else
+      extensionSignatureString +
+        s"def ${gsym.name}" +
+        paramssSignature
+
+  }
+
+  /*
+   * Check if a method is an extension method and in that case separate the parameters
+   * into 2 groups to make it possible to print extensions properly.
+   */
+  private def splitExtensionParamss(
+      gsym: Symbol
+  ): (List[List[Symbol]], List[List[Symbol]]) = {
+    def isUsingClause(params: List[Symbol]): Boolean = params match {
+      case sym :: _ => sym.is(Flags.Given)
+      case _ => false
+    }
+
+    def isTypeParamClause(params: List[Symbol]): Boolean = params match {
+      case sym :: _ => sym.is(Flags.TypeParam)
+      case _ => false
+    }
+
+    def isUsingOrTypeParamClause(params: List[Symbol]): Boolean =
+      isUsingClause(params) || isTypeParamClause(params)
+
+    val paramss =
+      if (gsym.rawParamss.length != 0) gsym.rawParamss else gsym.paramSymss
+    if (gsym.is(Flags.ExtensionMethod)) {
+      val filteredParams =
+        if (gsym.name.isRightAssocOperatorName) {
+          val (leadingTyParamss, rest1) = paramss.span(isTypeParamClause)
+          val (leadingUsing, rest2) = rest1.span(isUsingClause)
+          val (rightTyParamss, rest3) = rest2.span(isTypeParamClause)
+          val (rightParamss, rest4) = rest3.splitAt(1)
+          val (leftParamss, rest5) = rest4.splitAt(1)
+          val (trailingUsing, rest6) = rest5.span(isUsingClause)
+          if (leftParamss.nonEmpty)
+            leadingTyParamss ::: leadingUsing ::: leftParamss ::: trailingUsing ::: rightTyParamss ::: rightParamss ::: rest6
+          else paramss // it wasn't a binary operator, after all.
```

Comment:
Is there a test case on right accoc extension method?

Reply:
Not currently, I will add that.

Reply:
Added!